Sources/PackageLoading/ManifestLoader.swift

```diff
@@ -320,7 +320,7 @@ public final class ManifestLoader: ManifestLoaderProtocol {
             }
 
             // Check that products that reference only binary targets don't define a type.
-            let areTargetsBinary = product.targets.allSatisfy({ manifest.targetMap[$0]!.type == .binary })
+            let areTargetsBinary = product.targets.allSatisfy({ manifest.targetMap[$0]?.type == .binary })
```

Comment:
Can you write a test for this?

Reply:
There is a test. Its in the commit.

Reply:
Its called `testProductTargetNotFound`. Without this change, that test crashes.

Reply:
aah